### ISSUE TITLE
[syncfusion_flutter_maps] make map fill its viewport (visible area) – on initialization, zoom & pan (~Google Maps behavior)

### ISSUE THREAD

**Issue by timashev** [syncfusion/flutter-widgets#1494]
When I add a map like this:
```
child: SfMaps(layers: [MapTileLayer(urlTemplate: 'https://tile.openstreetmap.org/{z}/{x}/{y}.png')])
```
it is possible to pan or zoom the map to such position that one of its corners is in the middle of the screen, e.g.:

<img width="300" alt="image" src="https://github.com/syncfusion/flutter-widgets/assets/695076/17b33746-3e9d-474e-8963-ea77e3c55675">

or get such view initialy:

<img width="300" alt="image" src="https://github.com/syncfusion/flutter-widgets/assets/695076/90c160b2-171f-4dbc-b681-97232a99fafe">

**First thing,** I do not want the background to appear like this on the screen at all, I think that map should always cover all visible area of the parent widget – this is the expected behavior in most cases.

**Furthermore,** why limit visible longitude by min=-180 and max=+180 only? There must be an option of "cycle" the map and display it continuously by longitude, e.g. like [flutter_map](https://pub.dev/packages/flutter_map) does:

<img width="300" alt="image" src="https://github.com/syncfusion/flutter-widgets/assets/695076/b38cda0c-4911-44d5-8ae0-88197e71f315">

What is important in the "continuous" mode – display sublayers "cyclicly" as well, i.e. if there is a polygon over Africa, it must be displayed over all "copies" of Africas on the screen.

Regarding the **first problem** (map not covering all the viewport initially as well as on zoom/pan) – I tried to use `onWillPan` and `onWillZoom` callbacks like this:

```dart
bool _canPanZoom(MapLatLngBounds? previousVisibleBounds,
      MapLatLngBounds? newVisibleBounds) =>
  (newVisibleBounds?.northeast.longitude ?? 0) !=
      (previousVisibleBounds?.northeast.longitude ?? 0) &&
  (newVisibleBounds?.northeast.longitude ?? 0) < 179 &&
  (newVisibleBounds?.southwest.longitude ?? 0) !=
      (previousVisibleBounds?.southwest.longitude ?? 0) &&
  (newVisibleBounds?.southwest.longitude ?? 0) > -179 &&
  (newVisibleBounds?.northeast.latitude ?? 0) !=
      (previousVisibleBounds?.northeast.latitude ?? 0) &&
  (newVisibleBounds?.northeast.latitude ?? 0) < 84 &&
  (newVisibleBounds?.southwest.latitude ?? 0) !=
      (previousVisibleBounds?.southwest.latitude ?? 0) &&
  (newVisibleBounds?.southwest.latitude ?? 0) > -84;
...
  onWillPan: (details) => _canPanZoom(
      details.previousVisibleBounds, details.newVisibleBounds),
  onWillZoom: (details) => _canPanZoom(
      details.previousVisibleBounds, details.newVisibleBounds),
...
```

But this causes a kind of glitchy/twitching behavior. Additionally with these handlers it becomes impossible, e.g. to zoom out when one if the edges of the map is at ±84 / ±179 – the zoom is blocked by `onWillZoom` callback. The wanted behavior in this case is to let the zoom happen, but with simultaneous panning, i.e. the respective edge of the map should stay at its "edge" value (±84 / ±179) while zoom is increasing or decreasing. I found no way to implement this logic using existing callbacks!

So I think all the logic described above should be **implemented in `SfMaps` / `MapTileLayer` / `MapZoomPanBehavior` components as standard options**. Namely:

1. Option to constrain the map to always fill all the visible area, no background in the visible area.

2. Let the map be displayed "cyclicly" by longitude (see `flutter_map` screenshot above), and panning left-and-right without any limitations.

In short, this is how Google Maps work – I think it is a good example of proper world map behavior. Just implement the options to let the developers mimic Google Maps behavior, and `syncfusion_flutter_maps` will become the best map solution for Flutter! (it is good already, but `flutter_map` is still better in some use cases.)

**Comment by LokeshPalani** [comment]
Hi,

Currently, we do not have support for maps that should always cover all visible areas. We have already logged a feature request for it in our feedback portal.

We will prioritize the features of every release based on demand and priority. So, this feature will be available in any of our upcoming releases. You can also track the status of the feature with the feedback below.

Link - https://www.syncfusion.com/feedback/42719/provide-a-continuous-map-support-in-sfmaps

If you have any additional specifications or suggestions for this feature request, please feel free to leave them in the comments section of the feedback link. This will help us to understand how you would like to use it and how we can improve it.

Regards,
Lokesh Palani.
